sdk/digitaltwins/azure-digitaltwins/azure/digitaltwins/_digitaltwins_client.py

```diff
@@ -0,0 +1,547 @@
+        """Get a component on a digital twin.
+        Status codes:
+        200 (OK): Success.
+        404 (Not Found): There is either no digital twin with the provided id or the component path is
+        invalid.
+
+        :param digital_twin_id: The Id of the digital twin.
+        :type digital_twin_id: str
+        :param component_path: The component being retrieved.
+        :type str
+        :returns: The component object.
+        :rtype: object
+        :raises :class: `~azure.core.exceptions.HttpResponseError`
+        """
+        return self._client.digital_twins.get_component(digital_twin_id, component_path, **kwargs)
+
+    def update_component(
+        self,
+        digital_twin_id,
+        component_path,
+        component_patch,
+        etag=None,
+        **kwargs
+    ):
+        # type: (str, str, List[object],  str, **Any) -> object
+        """Update properties of a component on a digital twin using a JSON patch.
+        Status codes:
+        200 (OK): Success.
+        400 (Bad Request): The request is invalid.
+        404 (Not Found): There is either no digital twin with the provided id or the component path is
+        invalid.
+
+        :param digital_twin_id: The Id of the digital twin.
+        :type digital_twin_id: str
+        :param component_path: The component being updated.
+        :type str
+        :param component_patch: An update specification described by JSON Patch.
+        :type List[object]
+        :param etag: Only perform the operation if the entity's etag matches one of
+            the etags provided or * is provided.
+        :type str
+        :keyword str content_type: Content type to set in the HTTP header. Default to "application/json".
+        :returns: None
+        :rtype: None
+        :raises :class: `~azure.core.exceptions.HttpResponseError`
+        """
+        return self._client.digital_twins.update_component(
+            digital_twin_id,
+            component_path,
+            patch_document=component_patch,
+            if_match=etag,
+            **kwargs
+            )
+
+    def get_relationship(self, digital_twin_id, relationship_id, **kwargs):
+        # type: (str, str, **Any) -> object
+        """Get a relationship on a digital twin.
+        Status codes:
+        200 (OK): Success.
+        404 (Not Found): There is either no digital twin or relationship with the provided id.
+
+        :param digital_twin_id: The Id of the digital twin.
+        :type digital_twin_id: str
+        :param relationship_id: The Id of the relationship to retrieve.
+        :type str
+        :returns: The relationship object.
+        :rtype: object
+        :raises :class: `~azure.core.exceptions.HttpResponseError`
+        """
+        return self._client.digital_twins.get_relationship_by_id(digital_twin_id, relationship_id, **kwargs)
+
+    def upsert_relationship(self, digital_twin_id, relationship_id, relationship=None, **kwargs):
```

Comment:
What happens if I upsert a `None` relationtionship? Is this the same as deleting an existing relationship? Or reverting to defaults?

Reply:
It is going to be a placeholder for a relationship (with the given ID). It can be updated later.